[nccl] log rank when communicator is aborted

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57974 [nccl] log rank when communicator is aborted**

We see this error quite a bit in internal workflows, would be useful
to have this additional logging information here.

Differential Revision: [D28331693](https://our.internmc.facebook.com/intern/diff/D28331693/)

rohan-varma committed:
[nccl] log rank when communicator is aborted

We see this error quite a bit in internal workflows, would be useful
to have this additional logging information here.

Differential Revision: [D28331693](https://our.internmc.facebook.com/intern/diff/D28331693/)

[ghstack-poisoned]

SciPioneer commented on torch/lib/c10d/NCCLUtils.hpp:
```diff
@@ -112,6 +112,7 @@ class NCCLComm {
     C10D_NCCL_CHECK(
         ncclCommInitRank(&(comm->ncclComm_), numRanks, commId, rank));
     comm->ncclId_ = commId;
+    comm->rank_ = rank;
```
Just want to confirm: this will be global rank correct?

rohan-varma committed:
Update on "[nccl] log rank when communicator is aborted"


We see this error quite a bit in internal workflows, would be useful
to have this additional logging information here.

Differential Revision: [D28331693](https://our.internmc.facebook.com/intern/diff/D28331693/)

[ghstack-poisoned]
